### PR TITLE
Add support for CBOR

### LINKED ISSUE #31: Add optional CBOR support (low priority – can disable in Java SDK)
I am experiencing problems connecting to kinesalite using new versions of the AWS java sdk. The problem started after updating to version 1.11.0. The new versions of the sdk have no problem connecting to AWS kinesis.
Kinesalite is still working fine with the 1.10 versions.

The requests definitely get to the kinesalite, but the responses are malformed now.
Here is the error I get whenever I try to interact with kinesalite: 
`com.amazonaws.AmazonServiceException: Unable to parse HTTP response content (Service: AmazonKinesis; Status Code: 404; Error Code: null;`

### PR DESCRIPTION
**Pull request by tmccombs**
Fixes #31

### CODE REVIEW

**Comment by tmccombs** [comment]
I'd be happy to add additional test, but I'd like feedback on what the most valuable way to do that would be. I.e. Make most of the current tests run with both CBOR and JSON, just add some additional tests that use CBOR, or something else?

**Comment by alins** [comment]
any plans to have the conflicts resolved and get this PR to a conclusion? i think it would be awesome to have support for CBOR. 
also understand the low priority on this work since there is a work-around

**Comment by tmccombs** [comment]
I've resolved the merge conflicts.

**Comment by mhart** [comment]
Thanks so much @tmccombs – I'm looking at this now. Should be able to have it in sometime in the next couple of days

File: test/helpers.js [hunk 1]
```diff
@@ -82,9 +83,11 @@ function request(options, cb) {
       if ((res.headers['content-type'] || '').indexOf('application/x-amz-cbor') !== 0) {
         res.body = res.body.toString('utf8')
         try { res.body = JSON.parse(res.body) } catch (e) {} // eslint-disable-line no-empty
-        if (res.body.__type == 'LimitExceededException' && /^Rate exceeded/.test(res.body.message))
-          return setTimeout(request, Math.floor(Math.random() * 1000), options, cb)
```
Comment (mhart) [inline test/helpers.js]: Not sure why this has been removed?

File: test/helpers.js [hunk 2]
```diff
@@ -82,9 +83,11 @@ function request(options, cb) {
...
+      } else {
+        try { res.body = cbor.Decoder.decodeFirstSync(res.body) } catch (e) {} // eslint-disable-line no-empty
       }
+      if (res.body.__type == 'LimitExceededException' && /^Rate exceeded/.test(res.body.message))
+        return setTimeout(request, Math.floor(Math.random() * 1000), options, cb)
```
Comment (mhart) [inline test/helpers.js]: Ah, nevermind, missed this. My bad 👍

**Comment by mhart** [comment]
Merged and published as v3.3.0. Thanks so much for the contribution!

**Pull request merged**